### DISCUSSION TITLE
Splitting Grails 7 in subprojects (Divide Et Impera)

### DISCUSSION THREAD

**Discussion by gsartori** [apache/grails-core#13544] [Ideas]
It may be worth splitting Grails 7 release in subprojects each one of them with its own dedicated team.
Some of the subprojects must be sequential, some others may be run in parallel.

My ignorant guess:

SEQUENTIAL
x.7.1 - Porting Grails 6.2.0 to Groovy 4 + Java 17
x.7.2 - Removing Micronaut (can it become a plugin?)
x.7.3 - Refactoring the project architecture/structure + including the "core" plugins + adopt Gradle 8
x.7.4 - Porting to Spring Boot 3.3

PARALLEL
x.7.5 - Make each of the core plugins work
x.7.6 - Port GORM to "GroovyORM" with Hibernate 5.6 Jakarta release
x.7.7 - Make Spring Security Core Plugin work

I guess we then need a longer than usual testing/fixing phase.

**Comment by matrei** [comment]
There is a Github Project for managing the different tasks here: https://github.com/orgs/grails/projects/3

**Comment by sdelamo** [comment]
@gsartori I don't think we have several dedicated teams. Right now is just @matrei is pushing the development forward.

**Comment by jdaugherty** [comment]
Given where we are in the Grails 7 development, I believe this can be closed now.
